querier/querier.go

```diff
@@ -135,8 +140,39 @@ func (qm MergeQuerier) QueryInstant(ctx context.Context, ts model.Time, stalenes
 
 // MetricsForLabelMatchers Implements local.Querier.
 func (qm MergeQuerier) MetricsForLabelMatchers(ctx context.Context, from, through model.Time, matcherSets ...metric.LabelMatchers) ([]metric.Metric, error) {
-	// TODO: Implement.
-	return nil, nil
+	mss, errs := make(chan []metric.Metric), make(chan error)
+	for _, q := range qm.Queriers {
+		func(q Querier) {
```

Comment:
Yup!  I've abstracted out a function to do stuff in parallel across all ingesters, as we're doing this quite a lot...